### ISSUE TITLE
Docs: Versioned Quarkus Guides (Configuration etc)

### ISSUE THREAD

**Issue by epuronta** [keycloak/keycloak#12647]
### Description

For the new Quarkus-related documentation site at https://www.keycloak.org/guides, version-specific documentation is available but only the latest is deployed.

Now that the new distribution is marketed for production use, it is inevitable that older-than-latest versions are maintained for a non-trivial period of time.

Combine these, and there's bound to be people looking at documentation that applies to a version that's different to what they are actually running. Anecdotally, this just happened to me regarding the changes in the health endpoint (#10570).

The way administrative docs are published under https://www.keycloak.org/documentation makes perfect sense. Publish version-specific snapshots of the guides and have a version selector at the root of the guides page.

The other approach would be tracking the changes in the documents themselves ("KC_HEALTH_ENABLED added in 18.0.0"), but this tends to spiral out of control pretty quickly.

**Comment by sonOfRa** [comment]
Agreed, I also ran into this and started wondering if I just wasn't looking hard enough.

The server developer guides like https://www.keycloak.org/docs/19.0.3/server_development/index.html are still available (but also not easily, https://www.keycloak.org/archive/documentation-19.0.html links to https://www.keycloak.org/docs/19.0/server_development/index.html, which is a 404, I had to manually add the .3).

It would be very useful to have all documentation for older versions still available. In my experience, most people won't update their installations immediately on release day of a new version, and only having documentation available on the web for the most recent version makes it quite difficult to support versions that aren't even that old whenever a new version gets released.
